src/main/java/com/gluonhq/substrate/ProjectConfiguration.java

```diff
@@ -0,0 +1,246 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.substrate;
+
+import com.gluonhq.substrate.model.IosSigningConfiguration;
+import com.gluonhq.substrate.model.Triplet;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This class contains all public configuration info about the current project (not about the current OS/Arch/vendor etc)
+ * Instances of this class give hints to where real values should be contained from.
+ */
+public class ProjectConfiguration {
+
+    private String graalPath;
+
+    private String javafxStaticSdkVersion;
+    private String javaStaticSdkVersion;
+
+    private boolean usePrismSW = false;
+    private boolean verbose = false;
+
+    private Triplet targetTriplet;
+    private Triplet hostTriplet = Triplet.fromCurrentOS();
+
+    private List<String> bundlesList = Collections.emptyList();
+    private List<String> resourcesList = Collections.emptyList();
+    private List<String> reflectionList = Collections.emptyList();
+    private List<String> jniList = Collections.emptyList();
+
+    private String appName;
+    private String mainClassName;
+
+    private IosSigningConfiguration iosSigningConfiguration = new IosSigningConfiguration();
+
+    public ProjectConfiguration( String mainClassName ) {
+        this.mainClassName = Objects.requireNonNull(mainClassName, "Main class name is required")
+                               .contains("/") ?
+                                  mainClassName.substring( mainClassName.indexOf("/") + 1) : mainClassName;
+
+    }
+
+    public Path getGraalPath() {
+        return Path.of( Objects.requireNonNull(this.graalPath, "GraalVM Path is not defined"));
+    }
+
+    public void setGraalPath(String path) {
+        this.graalPath = path;
+    }
+
+
+    /**
+     * Sets the Java static SDK version
+     * @param javaStaticSdkVersion the Java static SDK version
+     */
+    public void setJavaStaticSdkVersion(String javaStaticSdkVersion) {
+        this.javaStaticSdkVersion = javaStaticSdkVersion;
+    }
+
+    public String getJavaStaticSdkVersion() {
+        return Objects.requireNonNull( this.javaStaticSdkVersion, "Java Static SDK version is required" );
+    }
+
+    /**
+     * Sets the JavaFX static SDK version
+     * @param javafxStaticSdkVersion the JavaFX static SDK version
+     */
+    public void setJavafxStaticSdkVersion(String javafxStaticSdkVersion) {
+        this.javafxStaticSdkVersion = javafxStaticSdkVersion;
+    }
+
+    public String getJavafxStaticSdkVersion() {
+        return Objects.requireNonNull( this.javafxStaticSdkVersion, "JavaFX Static SDK version is required" );
```

Comment:
same as above. In general, developers should be discouraged to set properties. The defaults should simply work.

Reply:
Fixed